File: src/chapter1/EQCount.java
package chapter1;

/**
 * Created on 16/5/8.
 * author: robinjia
 * email: dengshilong1988@gmail.com
 */
public class EQCount {
    public static int eqCount(int[] f, int[] g) {
        int i = 0;
        int j = 0;
        int result = 0;
        while (i < f.length && j < g.length) {
            if (f[i] == g[j]) {
                i++;
                j++;
                result++;
            } else if (f[i] > g[j]) {
                j++;
            } else {
                i++;
            }
        }
        return result;
    }
    public static void main(String[] args) {
        int[] f = {1, 3, 4, 7, 9};
        int[] g = {3, 5, 7, 8, 10};
        System.out.println(eqCount(f, g));
    }
}

File: chapter1/eq_count.py
# coding: utf-8


def eq_count(f, g):
    i = 0
    j = 0
    result = 0
    while i < len(f) and j < len(g):
        if f[i] == g[j]:
            i += 1
            j += 1
            result += 1
        elif f[i] > g[j]:
            j += 1
        else:
            i += 1
    return result


if __name__ == "__main__":
    f = [1, 3, 4, 7, 9]
    g = [3, 5, 7, 8, 10]
    print(eq_count(f, g))
